packages/css/src/types.ts

```diff
@@ -56,11 +56,25 @@ export interface ContainerQueries<StyleType> {
   };
 }
 
-export type WithQueries<StyleType> = MediaQueries<
+export interface Layer<StyleType> {
+  '@layer'?: {
+    [name: string]: StyleType;
+  };
+}
+
+export type WithQueries<StyleType> = Layer<
```

Comment:
This type says that layers can't be nested inside other conditional styles, but I believe it's possible to do this.